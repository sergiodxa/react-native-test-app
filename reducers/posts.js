import { Map as map } from 'immutable';

function posts(state = map(), action = {}) {
  switch (action.type) {
    case 'ADD_POST': {
      return state.set(action.payload.id, map(action.payload));
    }
    default: {
      return state;
    }
  }
}

export default posts;
