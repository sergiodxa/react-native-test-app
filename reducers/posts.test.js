import posts from './posts';

const action = {
  type: 'ADD_POST',
  payload: {
    id: 1,
    userId: 1,
    title: 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
    body: 'quia et suscipit suscipit recusandae consequuntur expedita et cum reprehenderit molestiae ut ut quas totam nostrum rerum est autem sunt rem eveniet architecto',
  },
};

describe('posts reducer', () => {
  it('should add a new post', () => {
    const state = posts(undefined, action);
    expect(state).toMatchSnapshot();
  });
});
