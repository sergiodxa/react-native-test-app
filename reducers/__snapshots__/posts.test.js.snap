// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts reducer should add a new post 1`] = `
Object {
  "1": Object {
    "body": "quia et suscipit suscipit recusandae consequuntur expedita et cum reprehenderit molestiae ut ut quas totam nostrum rerum est autem sunt rem eveniet architecto",
    "id": 1,
    "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
    "userId": 1,
  },
}
`;
